### PR TITLE
Improve night summary script

### PR DESCRIPTION
**Pull request by juanep97**
This PR improves the night summary script by:
 - using the same instrument color code in all plots. Previously, different plots in the same summary were choosing the colors independently, which was confusing when looking at the plots.
 - showing a warning if the specified return code of the iop4 command is not zero. Previously, a fatal error during the pipeline execution could result in the IOP4 script terminating prematurely and only some results showing in the summary script, without clear hints indicating that this happened.

### CODE REVIEW

File: iop4lib/iop4_night_summary.py [hunk 1]
```diff
@@ -71,6 +72,10 @@ def gather_context(args):
         
     sources = AstroSource.objects.exclude(is_calibrator=True).filter(photopolresults__epoch__night=args.date).distinct()
 
+    if sources:
+        instruments = list([instrument.name for instrument in Instrument.get_known()])
```
Comment (morcuended) [inline iop4lib/iop4_night_summary.py:76]: is list() needed?

Comment (juanep97) [reply iop4lib/iop4_night_summary.py]: nope

**Review by morcuended — DISMISSED** [review]
It looks good to me

File: iop4lib/iop4_night_summary.py [hunk 2]
```diff
@@ -201,6 +207,7 @@ def main():
     argparser.add_argument('--contact-email', type=str, default=None, help='Email to indicate as contact (default is the sender address)')
     argparser.add_argument('--site-url', type=str, default="localhost:8000", help='URL of the site to link the summary')
     argparser.add_argument('--saveto', type=str, default=None, help='Save the summary to a file')
+    argparser.add_argument('--rc', type=int, default=None, help="Indicate the return code to warn the user if the script fails")
```
Comment (morcuended) [inline iop4lib/iop4_night_summary.py:210]:
```suggestion
    argparser.add_argument('--rc', type=int, default=None, help="Indicate the return code from IOP4 to warn the user if the data processing fails")
```

**Review by morcuended — APPROVED** [review]

**Pull request merged**